src/components/Admin/AddQuestion.js

```diff
@@ -0,0 +1,259 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import { compose } from "recompose";
+
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+import { withFirebase } from "upe-react-components";
+
+import { isLoggedIn } from "../../util/conditions";
+import Loader from "../Loader";
+import Logo from "../Logo";
+import { Container } from "../../styles/global";
+
+export const RequiredAsterisk = styled.span`
+  color: red;
+  &:after {
+    content: "*";
+  }
+`;
+
+const StyledContainer = styled(Container)`
+  width: 300px;
+  padding: 15px;
+  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+  border-radius: 15px;
+  text-align: center;
+  &:hover {
+    -webkit-transform: translateY(-5px);
+    transform: translateY(-5px);
+    transition: all 0.3s linear;
+  }
+`;
+
+const CenteredForm = styled(Form)`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  max-width: 500px;
+  margin: 0 auto;
+`;
+
+class EditQuestion extends Component {
```

Comment:
This should also be a function component that gets data through props and doesnt track the data state. This component should be a form that gets a function passed as prop to save the data in firebase (so this component shouldn't directly access firebase). Basically the ManageQuestions.js should be the one tracking all the data state, and its children should get functions as props to modify the data. Also, instead of this should be inside of a modal on the parent.

Reply:
Resolved in commit 328335366bf582d828349cde311f9f15794bbca6